velox/functions/sparksql/tests/ArrayShuffleTest.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/VectorReaders.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+using namespace facebook::velox::test;
+
+class ArrayShuffleTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  VectorPtr
+  testShuffle(const VectorPtr& input, int64_t seed, int32_t partitionId = 0) {
+    setSparkPartitionId(partitionId);
+    return evaluate(
+        fmt::format("shuffle(c0, {})", seed), makeRowVector({input}));
+  }
+};
+
+TEST_F(ArrayShuffleTest, basic) {
+  auto input = makeArrayVector<int64_t>({{1, 2, 3, 4, 5}});
+  auto result = makeArrayVector<int64_t>({{3, 5, 4, 1, 2}});
+  auto resultForPartitionIdOne = makeArrayVector<int64_t>({{2, 1, 3, 4, 5}});
```

Comment:
The names of resultForPartitionIdOne and resultForSeesTwo are long and a bit hard to reason about. Consider removing them altogether.

```
assertEqualVectors(
      testShuffle<int64_t>(input, 0, 1), makeArrayVector<int64_t>({{2, 1, 3, 4, 5}}));

assertEqualVectors(
      testShuffle<int64_t>(input, 2, 0), makeArrayVector<int64_t>({{4, 1, 3, 5, 2}}));
```

Alternatively, we can enhance testShuffle to take 2 JSON strings for input and result arrays:

```
testShuffle<int64_t>("[1, 2, 3, 4, 5]", 0, 1), "[2, 1, 3, 4, 5]");
```

Reply:
Thanks.